inspirehep/modules/workflows/tasks/actions.py

```diff
@@ -827,3 +829,78 @@ def extract_authors_from_pdf(obj, eng):
                 grobid_authors_count
             )
     return obj
+
+
+def prepare_collaboration_multi_search(collaborations):
+    multi_search = MultiSearch(index="records-experiments", using=current_search_client)
+    backup_multi_search = MultiSearch(index="records-experiments", using=current_search_client)
+    for collaboration in collaborations:
+        full_collaboration_string = collaboration.get('value', '')
+        normalized_collaboration_string = normalize_collaboration_name(full_collaboration_string)
+        if collaboration.get('record'):
+            # Add dummy search so multisearch will stay in sync with collaborations
+            multi_search = multi_search.add(Search().query().source(False))
+            backup_multi_search = backup_multi_search.add(Search().query().source(False))
+            continue
+
+        multi_search = multi_search.add(build_standard_collaboration_search(normalized_collaboration_string))
+        backup_multi_search = backup_multi_search.add(build_subgroup_collaboration_search(normalized_collaboration_string))
+    return multi_search, backup_multi_search
+
+
+def build_standard_collaboration_search(normalized_collaboration_string):
+    name_search = Q("term", normalized_name_variants={"value": normalized_collaboration_string})
+
+    query = name_search
+    filters_ = Q("exists", field="collaboration")
+    return Search().query(query).filter(filters_).source(['collaboration', 'self', 'legacy_name'])
+
+
+def build_subgroup_collaboration_search(normalized_collaboration_string):
+    subgroup_search = Q("term", normalized_subgroups={"value": normalized_collaboration_string})
```

Comment:
Minor: only this line changes from the previous function. Maybe you could combine them somehow?

Reply:
:+1: